### PR TITLE
Add possible set password without encryption

### PR DESCRIPTION
**Pull request by ByJacob**
## what
Allowing the user to create an iam account with an unencrypted password

## why

- easy password creation
- the password can be created without the need for the pgp private key

## references

Don't close issue :)

### CODE REVIEW

**Comment by ByJacob** [comment]
@Gowiem can we process this MR ?

File: README.yaml [hunk 1]
```diff
@@ -83,6 +83,18 @@ examples: |-
...
+  ```hcl
+  module "jill" {
+    source = "cloudposse/iam-user/aws"
+    # Cloud Posse recommends pinning every module to a specific version
+    # version     = "x.x.x"
+    name      = "jill"
+    user_name  = "jill@companyname.com"
+    groups    = ["engineering"]
+    password_encrypted = false
+    decrypted_password = "P@$w0rd"
```
Comment (nitrocode) [inline README.yaml:95]:
```suggestion
    name               = "jill"
    user_name          = "jill@companyname.com"
    groups             = ["engineering"]
    password_encrypted = false
    decrypted_password = "P@$w0rd"
```

File: main.tf [hunk 2]
```diff
@@ -18,6 +18,41 @@ resource "aws_iam_user_login_profile" "default" {
...
+locals {
+  aws_cli_command         = length(var.aws_cli_profile) > 0 ? "${var.aws_cli_command} --profile ${var.aws_cli_profile}" : var.aws_cli_command
+  user_password           = length(var.decrypted_password) > 0 ? var.decrypted_password : join("", random_password.password.*.result)
+  user_password_file_path = "${path.module}/${var.user_name}_password.txt"
+  password_reset_required = var.password_reset_required ? "--password-reset-required" : "--no-password-reset-required"
+  set_password_script     = <<EOF
+  #!/bin/bash
+  ${local.aws_cli_command} iam create-login-profile --user-name ${var.user_name} --password ${local.user_password} ${local.password_reset_required}
+EOF
+}
```
Comment (nitrocode) [inline main.tf:36]: Can we move these locals to the top of the terraform ?

File: main.tf [hunk 3]
```diff
@@ -18,6 +18,41 @@ resource "aws_iam_user_login_profile" "default" {
...
+  }
+
+  provisioner "local-exec" {
+    command     = local.set_password_script
+    interpreter = var.bash_interpreter
+  }
+}
+
+resource "local_file" "password_file" {
+  content  = local.user_password
```
Comment (nitrocode) [inline main.tf:52]: Count logic is missing here

File: main.tf [hunk 4]
```diff
@@ -18,6 +18,41 @@ resource "aws_iam_user_login_profile" "default" {
...
+  user_password           = length(var.decrypted_password) > 0 ? var.decrypted_password : join("", random_password.password.*.result)
+  user_password_file_path = "${path.module}/${var.user_name}_password.txt"
+  password_reset_required = var.password_reset_required ? "--password-reset-required" : "--no-password-reset-required"
+  set_password_script     = <<EOF
+  #!/bin/bash
+  ${local.aws_cli_command} iam create-login-profile --user-name ${var.user_name} --password ${local.user_password} ${local.password_reset_required}
+EOF
+}
+
+resource "null_resource" "deploy" {
```
Comment (nitrocode) [inline main.tf:38]: Is this the only way to do this ?

**Comment by nitrocode** [comment]
This seems unsafe to me and we usually discourage using a `null_resource` techniques.

What is the reason to want to set a password without encryption ?

**Comment by Gowiem** [comment]
@ByJacob friendly on this PR -- Want to keep it updated or close it?

**Comment by ByJacob** [comment]
> @ByJacob friendly ping on this PR -- Want to keep it updated or close it?

Unfortunately, until closing. I don't need this feature anymore, and I see the conversation going on in the thread https://github.com/hashicorp/terraform-provider-aws/issues/18749

**Pull request closed without merging**